gateway/gateway.c

```diff
@@ -1,30 +1,110 @@
 #include "../sx1302/libloragw/inc/loragw_hal.h"
 #include <stdint.h>
 #include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <unistd.h>
 
 #define MAX_RX_PKT 8
 
-int setBoardConfig(struct lgw_conf_board_s conf) {
-    return lgw_board_setconf(&conf);
-}
+#define RADIO_0_FREQ     902700000
+#define RADIO_1_FREQ     903700000
+#define RX2_FREQ  923300000
+#define RX2_SF  12
+#define RX2_BANDWIDTH 0x06
 
-int setRFConfig(uint8_t rfchain, struct lgw_conf_rxrf_s conf) {
-    	return lgw_rxrf_setconf(rfchain, &conf);
-}
 
-int setIFConfig(uint8_t ifChain, struct lgw_conf_rxif_s conf) {
-    return lgw_rxif_setconf(ifChain, &conf);
-}
+// the IF chain frequencies allow the gateway to read on multiple frequency channels.
+// subtracting main frequenecy - intermediate frequency will give that channel's freq.
+const int32_t ifFrequencies[9] = {
+	-400000,
+	-200000,
+	0,
+	200000,
+	400000,
+	-400000,
+	-200000,
+	0,
+};
+
+// This defines what RF chain to use for each of the 8 if chains
+const int32_t rfChains [9] = {0, 0, 0, 0, 1, 1, 1};
+
+
+int setUpGateway(int bus) {
+    struct lgw_conf_board_s boardconf;
+    struct lgw_conf_rxrf_s rfconf;
+    struct lgw_conf_rxif_s ifconf;
```

Comment:
what's the difference between rfconf and ifconf? I'm sure this (and many of my questions) will be because I'm not familiar with the hardware, so maybe add a link to helpful documentation if it exists?